### PR TITLE
add documentation to show how to improve compilation time

### PR DESCRIPTION
**Pull request by pmnoxx**
This is a preview of documentation, which shows how to improve compilation time of building nearcode.

### CODE REVIEW

File: docs/contribution/nearcore.md [hunk 1]
```diff
@@ -114,3 +114,108 @@ Official image is published at `nearprotocol/nearcore`
...
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6' RUSTC_WRAPPER= cargo build -p neard --release
+real	4m42.485s user	28m47.223s sys	2m18.833s
+# touch */*/*/*.rs; time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6' RUSTC_WRAPPER= cargo build -p neard --release
+real	0m51.620s user	4m13.180s sys	0m11.254s
+```
+
+# cache results from previous compilations using sccache
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6' RUSTC_WRAPPER=sccache cargo build -p neard --release
```
Comment (bowenwang1996) [inline docs/contribution/nearcore.md:170]: this one seems to be exactly the same as the command below

File: docs/contribution/nearcore.md [hunk 2]
```diff
@@ -114,3 +114,108 @@ Official image is published at `nearprotocol/nearcore`
...
+```
+
+For other systems follow link at https://lld.llvm.org/
+If you can't install lld, you can remove option `-C link-arg=-fuse-ld=lld`.
+
+
+## Clion/Intellij with Rust plugin
+Currently Rust plugin doesn't support passing enviroment variables to cargo.
+Your best option is to either modify Cargo.toml or replace
+`/home/pmnoxx/.cargo/bin/cargo` with a bash script, which sets proper
```
Comment (bowenwang1996) [inline docs/contribution/nearcore.md:220]: probably make sense to hide your username?

File: docs/contribution/nearcore.md [hunk 3]
```diff
@@ -114,3 +114,108 @@ Official image is published at `nearprotocol/nearcore`
...
+
+```
+
+# Setting for building release binary with debug symbols and reduced inlining
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6 -C inline-threshold=25 -C debuginfo=2' RUSTC_WRAPPER=sccache cargo build -p neard --release
+real	4m3.216s user	6m2.943s sys	0m54.196s
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6 -C inline-threshold=25 -C debuginfo=2' RUSTC_WRAPPER=sccache cargo build -p neard --release
+real	1m30.039s user	5m50.652s sys	1m2.509s
+# touch */*/*/*.rs; time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6 -C inline-threshold=25 -C debuginfo=2' RUSTC_WRAPPER=sccache cargo build -p neard --release
```
Comment (bowenwang1996) [inline docs/contribution/nearcore.md:195]: seems that there is no difference in these commands?

File: docs/contribution/nearcore.md (hunk 2 above)
Comment (pmnoxx) [reply docs/contribution/nearcore.md]: fixed

File: docs/contribution/nearcore.md (hunk 3 above)
Comment (pmnoxx) [reply docs/contribution/nearcore.md]: fixed

File: docs/contribution/nearcore.md (hunk 1 above)
Comment (bowenwang1996) [reply docs/contribution/nearcore.md:174]: how is this resolved?

Comment (pmnoxx) [reply docs/contribution/nearcore.md:174]: It's fixed now..

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**